src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormattingPassBase.cs

```diff
@@ -171,6 +171,19 @@ protected async Task<List<TextChange>> AdjustIndentationAsync(FormattingContext
                     // We use binary search to find that spot.
 
                     var index = Array.BinarySearch(sourceMappingIndentationScopes, lineStart);
+
+                    if (index < 0 && context.SourceText[lineStart] == '@')
```

Comment:
Interesting, I don't know if I fully follow the tactical fix (most likely because I'm lacking external context). Would love if you could run me through it!

Reply:
Given this code, with `<#` and `#>` being the source mappings.

```
@section <#Scripts#> {
  <p></p>
}

@<#if (true) {
}#>
```

This code goes through each line, and calculates what the indention would be in the generated C# code, for the start of each line, and then use that information to work out what the indentation should be in the razor file. When it can't find a direct mapping to the generated C# (which it can't at the start of line 5, the `@if`) it uses the previous known source mapping, and assumes the generated C# at that point has the same indentation as the point we're interested in.

In the case of `@section`s there is no source mapping at the end of the section block, so the previous known source mapping ends up being the one for the section name, and that comes back with an indentation that includes the indentation caused by the lambda that renders the section.

Does that make sense?

Reply:
ooooooooooo I totallly get it, thank you for clarifying!

As a follow up, is there a reason why we jump multiple lines looking for another source mapping? Does that typically yield reasonable results? I guess I question "finding the last line mapping" when the differnece in line positions is > 1

Reply:
This is only ever trying to find the indentation for the start of each line, so not looking at other lines is effectively removing this, and there are a number of tests that fail when that happens. "Why" is always a fun question though :)

I suspect the "real" solution here would be for the compiler to include a source mapping for the start of ever line, as a matter of course, though whether it has the info to do that, or what else that breaks, is anyones guess :)